### ISSUE TITLE
EF Core Closed Connection

### ISSUE THREAD

**Issue by Wayne-Mather** [dotnet/efcore#26339]
Hi,

I am on a project using .NET Core 3.1LTS. We are using context.Database.BeginTransaction() to create a transaction. We are also using EFCore.BulkExtensions and are using TempDB for the inserts.

The issue we have is that while the code is in the transaction scope, it looks like EF has closed the connection. If we query context.Database.GetDbConnection().State , it seems the state has already closed (even through the transaction has not been commited or disposed).

Has anyone found this issue and/or a workaround? I know maybe hard to understand my question with no code, as I cannot put the code into the public domain without violation of my contract obligations and occurring penalties.

Just wondering if anyone else has come across this issue before.

**Comment by roji** [comment]
@Wayne-Mather there isn't enough information above for us to investigate this - we're going to need some sort of code sample. In this sample, at some point between BeginTransaction and your BulkInsert, the transaction is somehow closed - we'd need to understand what you're doing to trigger that.

> I would assume that GetDbConnection() would return the current underlying connection and that there would be one for each DbContext class instantiated? NOTE: This is an a .NET Core 3.1 WebApi application.

Each DbContext indeed has its own DbConnection. However, the DbConnection is only kept open for the minimum amount of time needed - and not throughout the lifetime of the context. When not using connections, the connection is opened when a query is executed and closed directly afterwards. When using a transaction, the connection is kept open for the lifetime of the transaction.

**Comment by Wayne-Mather** [comment]
Going to close this ticket here. From all my testing I do not believe it's an issue with EF Core. The library in question does close any connections, therefore a transaction will have issues across multiple calls.

TLDR; Issue with either 3rd-Party code or project code - but not within the EF Core library
